### ISSUE TITLE
Incorrect error with nested array destructuring pattern

### ISSUE THREAD

**Issue by danquirk** [Microsoft/TypeScript#2089]
```ts
var [x,[a,b]] = [[]]; // error
var r = [[]];
var [x,[a,b]] = r; // ok
```
a.ts(5,8): error TS2460: Type '[undefined[]]' has no property '1'.

Seems to be a widening issue?

**Comment by danquirk** [comment]
By design, gonna log a separate issue for error messages
